modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/SpringComponentModel.java

```diff
@@ -80,6 +95,80 @@ public ComponentLocation getLocation() {
     }
   }
 
+  /**
+   * @param parameterName name of the configuration parameter.
+   * @param value value contained by the configuration parameter.
+   */
+  @Override
+  public void setParameter(ParameterModel parameterModel, ComponentParameterAst value) {
+    super.setParameter(parameterModel, value);
+    this.parameterAsts.put(parameterModel.getName(), value);
+  }
+
+  /**
+   * @param paramName the name of the parameter to get AST for.
+   * @return the AST of the parameter if present, or {@link Optional#empty()} if not present.
+   */
+  @Override
+  public ComponentParameterAst getParameter(String paramName) {
+    return parameterAsts.computeIfAbsent(paramName, paramNameKey -> findParameterModel(paramNameKey)
+        .map(paramModel -> getRawParameterValue(paramNameKey)
+            .map(rawParamValue -> new DefaultComponentParameterAst(rawParamValue, () -> paramModel))
+            .orElseGet(() -> new DefaultComponentParameterAst(null, () -> paramModel)))
+        .orElseThrow(() -> {
+          // Try to provide as much troubleshooting information as possible
+          final String mxgPrefix = format("Wanted paramName '%s' from object '%s'.", paramNameKey,
+                                          getModel(NamedObject.class).map(n -> n.getName()).orElse("(null)"));
+
+          if (!getModel(ParameterizedModel.class).isPresent()) {
+            return new NoSuchElementException(mxgPrefix + " The model is not parameterizable.");
+          } else {
+            final List<String> availableParams = getModel(ParameterizedModel.class).get().getAllParameterModels()
+                .stream()
+                .map(pm -> pm.getName()).collect(toList());
+            return new NoSuchElementException(mxgPrefix + " Available: " + availableParams);
+          }
+        }));
+  }
+
+  private Optional<ParameterModel> findParameterModel(String paramName) {
+    // For sources, we need to account for the case where parameters in the callbacks may have colliding names.
+    // This logic ensures that the parameter fetching logic is consistent with the logic that handles this scenario in previous
+    // implementations.
+    Optional<ParameterModel> parameterModel = getModel(SourceModel.class)
```

Comment:
@ndinu we should review this as it is related to the decision made about allowing to have repeated names on source callbacks